producers/mysql-producer/src/main/java/com/flipkart/aesop/runtime/producer/avro/MysqlAvroEventManager.java

```diff
@@ -168,6 +194,31 @@ public int createAndAppendEvent(DbChangeEntry changeEntry, DbusEventBufferAppend
 	}
 
 	/**
+	 * This method scans new and old records and calculates changes.
+	 *
+	 * @param oldRecord is a generic record representing old-record before change
+	 * @param newRecord is a generic record representing new-record after change
+	 * @return HashMap<String, Object> this contains changes in form of <columnName,ColumnValue>
+	 */
+	private static Map<String, Object> calculateChange(GenericRecord oldRecord, GenericRecord newRecord)
+	{
+		Map <String, Object> changedOldValue = new HashMap<String, Object>();
+		for (Schema.Field  field : newRecord.getSchema().getFields())
+		{
+			String fieldName = field.name();
+			Object oldValue = oldRecord.get(fieldName);
+			Object newValue = newRecord.get(fieldName);
+
+			if (!Objects.equal(oldValue, newValue))
+			{
+				changedOldValue.put(fieldName, oldValue);
+			}
+		}
+		assert (changedOldValue.isEmpty() == false) : "Old and New Record values are same or equals not working as expected";
```

Comment:
Avro schema currently is tightly coupled with DB, so it will always have all the columns in table.
If avro does not have all columns (less than table), then event creation in mysql producer will work eratically or randomly.